### ISSUE TITLE
findById, modify and then save cause unique validation of uchanged fields?

### ISSUE THREAD

**Issue by lijax** [blakehaswell/mongoose-unique-validator#43]
Here is my code. And this cause validation of all unique fields in the UserSchema.

```
  User.findById(userId)
    .then(user => {
      user.app.secret = hat(128, 16);
      return user.save()
        .then(data => {
          apiTool.created(req, res, data);
        })
        .catch(validationError(req, res));
    });
```

**Comment by viveleroi** [comment]
What version are you using?

**Comment by lijax** [comment]
mongodb: 2.4.9
mongoose: 4.3.6
mongoose-unique-validator: 1.0.1

**Comment by viveleroi** [comment]
This fix is included in version 1.0.2

**Comment by lijax** [comment]
yeah, fixed. Thanks!
